jme3-core/src/main/java/com/jme3/animation/SkeletonControl.java

```diff
@@ -210,15 +214,23 @@ public SkeletonControl(Skeleton skeleton) {
         this.skeleton = skeleton;
     }
 
+    /**
+     * If specified the geometry has an animated mesh, add its mesh and material
+     * to the lists of animation targets.
+     */
+    private void findTargets(Geometry geometry) {
+        Mesh mesh = geometry.getMesh();
+        if (mesh.isAnimated()) {
+            targets.add(mesh);
+            targetGeometry = geometry;
```

Comment:
Mhh this won't work. Here you are using the last geometry in the subgraph for every bones.
You need to read the BoneIndex buffer to check if the bone affects this geometry. This should be done in the Bone class